src/components/header.jsx

```diff
@@ -111,48 +107,42 @@ const NavLink = styled.a`
   }
 `;
 
-const Header = (props) => {
-  const { defaultSelected, router } = props;
+const Header = ({ defaultSelected, router }) => {
   const classes = useStyles();
+  const [session, loading] = useSession();
+
   const [anchorEl, setAnchorEl] = React.useState(null);
   const [selected, setSelected] = React.useState(defaultSelected);
   const [filteredRoutes, setFilteredRoutes] = React.useState([]);
-  const [session, loading] = useSession();
 
-  // console.log(currentUser);
-  // let filteredRoutes = [];
-  // const [filteredRoutes, setFilteredRoutes] = React.useState([]);
+  React.useEffect(() => async () => {
+    if (!loading && session && filteredRoutes.length == 0) {
+      const json = await queryUser();
+      setFilteredRoutes(filterRoutes(json.payload));
+    }
+  });
 
   const filterRoutes = (currentUser) => {
-    let fRoutes = [];
-    if (currentUser.type == "Admin") {
-      fRoutes = routes.filter(
-        (item) => item.type == "Admin" || item.type == "All"
-      );
-    } else if (currentUser.type == "BusDriver") {
-      fRoutes = routes.filter(
-        (item) => item.type == "BusDriver" || item.type == "All"
-      );
-    } else if (currentUser.type == "ClubDirector") {
-      fRoutes = routes.filter(
-        (item) =>
-          item.type == "ClubDirectorAttendanceClerk" || item.type == "All"
-      );
-    } else if (currentUser.type == "AttendanceClerk") {
-      fRoutes = routes.filter(
-        (item) =>
-          item.type == "ClubDirectorAttendanceClerk" || item.type == "All"
-      );
-    } else {
-      fRoutes = routes.filter((item) => item.type == "All");
+    let allRoutes = routes.filter(item => item.type === "All");
+
+    let userRoutes = [];
+    if (currentUser.type === "Admin") {
+      userRoutes = routes.filter(item => item.type === "Admin");
+
+    } else if (currentUser.type === "BusDriver") {
+      userRoutes = routes.filter(item => item.type === "BusDriver");
+
+    } else if (currentUser.type === "ClubDirector" || currentUser.type === "AttendanceClerk") {
+      userRoutes = routes.filter(item => item.type === "ClubDirectorAttendanceClerk");
     }
 
-    return fRoutes;
+    return allRoutes.concat(userRoutes);
   };
+
   const queryUser = async () => {
     console.log("QUERY USER", session.user.email);
-    const res = await fetch(`/api/user?email=${session.user}`);
-    return await res.json();
+    const res = await fetch(`${urls.api.user}?email=${session.user.email}`);
```

Comment:
Root cause of the bug was a miscommunication in the way that the API expected the email to be passed and the manner in which our client was including it in the request.